whats_fresh/whats_fresh_api/templates/vendor.html

```diff
@@ -256,23 +258,80 @@ <h3 class="title">Products</h3>
 
     function setAddressFields()          // For determining the location parameters.
     {
-
         $('input[name="street"]').val($('#map').data("locationpicker").addressComponents.addressLine1);
         $('input[name="city"]').val($('#map').data("locationpicker").addressComponents.city);
         $('input[name="state"]').val($('#map').data("locationpicker").addressComponents.stateOrProvince);
         $('input[name="zip"]').val($('#map').data("locationpicker").addressComponents.postalCode);
-
     }
 
-
     var preparation_options = {{ json_preparations|safe }};
 
     number_of_products = 0;
 
+    function popitup(url) {
+      $.ajax({
+        type: "GET",
+        url: url,
+        headers: { 'X-CSRFToken': '{{ csrf_token }}' },
+        success: function(data) {
+
+          $("#dialog-form").html(data).dialog({modal:true}, {title: "New Product"}, {resizable: true}, {width: 'auto'}, {draggable: false}).dialog('open');
+        }
+      });
+      return false;
+    }
+
+
+   $("#dialog-form").on('submit', function (e) {
+
+     e.preventDefault();
+
+     $.ajax({
+        type: "POST",
+        url: '/entry/vendors/new/products/new',
+        data: $("#popup-form").serialize(),
+        headers: { 'X-CSRFToken': '{{ csrf_token }}' },
+        success: function(data)
+        {
+            if (data.constructor === "test".constructor) {     // checks if the data is a string
+                $("#dialog-form").html(data);
+                $('html, body').animate({
+                    scrollTop: $("#product_error").offset().top - 100
+                }, 500);
+                return
+            }
+
+            $("#dialog-form").dialog('close');
+            var prodName = data.name;
+            var prodId = data.id;
+
+            preparation_options[prodName] = [];
+
+            for(i=0; i < data.preparations.length; i++) {
+
+                preparation_options[prodName].push({name: data.preparations[i].name, value: data.preparations[i].id});
+
+            }
+
+            $('#product_select').append($("<option>").attr("value", prodName).html(prodName));
+
+            appendProduct();
+
+            $("#product" + (number_of_products - 1) + " option").filter(function() {
+                return $(this).text() == prodName;
+            }).prop('selected', true);
+
+            showPreparations(number_of_products - 1);
+
+        }
+      });
+   });
+
+
     function deleteProduct(element)
     {
         $( element ).parent().css("display", "none");
-        if ($( ".preparation:visible" ).length === 0) {
+        if ($( ".preparation:visible" ).length === 0 && $( ".new_product:visible" ).length === 0 ) {
```

Comment:
Why is this selecting .preparation and .new_product? Actually, I think this is the bug that I mentioned earlier, shouldn't this look for products not preparations? :tongue: 

Same as above, I'd give the new products the same class as the others.